dot/state/block.go

```diff
@@ -250,14 +250,62 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetAllDescendants gets all the descendants even if hash is not stored in memory.
+func (bs *BlockState) GetAllDescendants(hash common.Hash) ([]common.Hash, error) {
+	allDescendants, err := bs.bt.GetAllDescendants(hash)
+	if err != nil && !errors.Is(err, blocktree.ErrNodeNotFound) {
+		return nil, err
+	}
+
+	if err == nil {
+		return allDescendants, nil
+	}
+
+	header, err := bs.GetHeader(hash)
+	if err != nil {
+		return nil, fmt.Errorf("getting header from hash %s: %w", hash, err)
+	}
+
+	// TODO: Use GetBlocksByNumber after https://github.com/ChainSafe/gossamer/issues/2748 is done
+	nextBlock, err := bs.GetBlockByNumber(header.Number + 1)
+	if err != nil {
+		return nil, fmt.Errorf("getting block by number %d: %w", header.Number+1, err)
+	}
+
+	// next block number is not descendant of hash
```

Comment:
It's not the next block number really, it's the next block that is descendant

```suggestion
	// next block is not a descendant of the block for the given hash
```